src/frontend/src/lib/components/guard/AddressGuard.svelte

```diff
@@ -1,29 +1,49 @@
 <script lang="ts">
 	import { addressStore } from '$lib/stores/address.store';
-	import { isNullish } from '@dfinity/utils';
-	import { certifyEthAddress } from '$lib/services/address.services';
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { certifyBtcAddressMainnet, certifyEthAddress } from '$lib/services/address.services';
 	import { warnSignOut } from '$lib/services/auth.services';
-	import { ETHEREUM_TOKEN_ID } from '$env/tokens.env';
+	import {
+		btcAddressMainnet,
+		btcAddressMainnetData,
+		btcAddressMainnetNotCertified,
+		ethAddress,
+		ethAddressData,
+		ethAddressNotCertified
+	} from '$lib/derived/address.derived';
+	import { NETWORK_BITCOIN_ENABLED } from '$env/networks.btc.env.js';
+	import type { ResultSuccess } from '$lib/types/utils';
 
 	const validateAddress = async () => {
-		const ethAddressData = $addressStore?.[ETHEREUM_TOKEN_ID];
-
-		if (isNullish(ethAddressData)) {
+		if (isNullish($btcAddressMainnetData) && isNullish($ethAddressData)) {
 			// No address is loaded, we don't have to verify it
 			return;
 		}
 
-		const { certified, data: ethAddress } = ethAddressData;
-
-		if (certified === true) {
-			// The address is certified, all good
+		if ((!NETWORK_BITCOIN_ENABLED || !$btcAddressMainnetNotCertified) && !$ethAddressNotCertified) {
```

Comment:
this hurts my brain 🤪. in another PR maybe we should transform it into a "positive flag check" assertion?